### ISSUE TITLE
Classification Schema: no way to get the list of ifc types

### ISSUE THREAD

**Issue by Fredosixx** [SketchUp/api-issue-tracker#1020]
I did not find **a way to get the list of ifc types in a classification schema**, I mean this list:

![image](https://github.com/user-attachments/assets/9c8b8a97-4806-4b1c-84cc-12046e34a2a9)

The class ClassificationSchema only allows to get the name (ex: IFC 2x3) or the namespace URL (e.g. "http://www.iai-tech.org/ifcXML/IFC2x3/FINAL"), nothing else.

Am I missing anything?

**Comment by rafhik** [comment]
Fredo you can try to rename the schema file to .zip then unzip it and you will find several xml files with all classes.
Is that what you need?
Sketchup read a file inside Sketchup folder named IFC4.skc and IFC 2x3.skc

**Comment by rafhik** [comment]
You con read how @janbrouwer export Ifc files with his Ifc Manager SketchUp plugin too...
https://github.com/BIM-Tools/SketchUp-IFC-Manager

**Comment by janbrouwer** [comment]
Yes, I had the same issue. I included ifczip and parse the xml myself. Not ideal...
https://github.com/BIM-Tools/SketchUp-IFC-Manager/blob/main/src%2Fbt_ifcmanager%2Flib%2Flib_ifc%2Fparse_xsd.rb

**Comment by Fredosixx** [comment]
Thanks. I understand why some user say that Sketchup is not very IFC-friendly...!

**Comment by janbrouwer** [comment]
I really do like the concept of SKC. For IFC it directly includes the official XSD schema. But besides that it allows very generic model data enrichment. Sadly it was never really adopted by many people. That's probably also why the API is lacking. I see many possibilities using the bSDD (buildingSMART Data Dictionary) and SKC. @thomthom knows a lot about it
